Gems/Atom/Tools/AtomToolsFramework/Code/Source/AssetSelection/AssetSelectionGrid.cpp

```diff
@@ -196,15 +122,78 @@ namespace AtomToolsFramework
         AzQtComponents::Text::addLabelStyle(header);
         itemWidget->layout()->addWidget(header);
 
+        bool result = false;
+        AZ::Data::AssetInfo assetInfo;
+        AZStd::string watchFolder;
+        AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+            result, &AzToolsFramework::AssetSystemRequestBus::Events::GetSourceInfoBySourcePath, path.data(), assetInfo, watchFolder);
+
         AzToolsFramework::Thumbnailer::ThumbnailWidget* thumbnail = new AzToolsFramework::Thumbnailer::ThumbnailWidget(itemWidget);
         thumbnail->setFixedSize(m_tileSize);
-        thumbnail->SetThumbnailKey(MAKE_TKEY(AzToolsFramework::AssetBrowser::ProductThumbnailKey, assetId));
+        thumbnail->SetThumbnailKey(MAKE_TKEY(AzToolsFramework::AssetBrowser::SourceThumbnailKey, assetInfo.m_assetId.m_guid));
         thumbnail->updateGeometry();
         itemWidget->layout()->addWidget(thumbnail);
 
         m_ui->m_assetList->setItemWidget(item, itemWidget);
+        m_ui->m_assetList->sortItems();
+    }
 
-        return item;
+    void AssetSelectionGrid::RemovePath(const AZStd::string& path)
+    {
+        const QVariant pathItemData(QString::fromUtf8(path.data(), static_cast<int>(path.size())));
+        for (int i = 0; i < m_ui->m_assetList->count(); ++i)
+        {
+            QListWidgetItem* item = m_ui->m_assetList->item(i);
+            if (pathItemData == item->data(Qt::UserRole))
+            {
+                m_ui->m_assetList->removeItemWidget(item);
+                return;
+            }
+        }
+    }
+
+    void AssetSelectionGrid::SelectPath(const AZStd::string& path)
+    {
+        const QVariant pathItemData(QString::fromUtf8(path.data(), static_cast<int>(path.size())));
+        for (int i = 0; i < m_ui->m_assetList->count(); ++i)
+        {
+            QListWidgetItem* item = m_ui->m_assetList->item(i);
+            if (pathItemData == item->data(Qt::UserRole))
+            {
+                m_ui->m_assetList->setCurrentItem(item);
+                return;
+            }
+        }
+    }
+
+    AZStd::string AssetSelectionGrid::GetSelectedPath() const
+    {
+        auto item = m_ui->m_assetList->currentItem();
+        return item ? AZStd::string(item->data(Qt::UserRole).toString().toUtf8().constData()) : AZStd::string();
+    }
+
+    void AssetSelectionGrid::OnCatalogAssetAdded(const AZ::Data::AssetId& assetId)
+    {
+        if (m_filterCallback)
+        {
+            const auto& path = AZ::RPI::AssetUtils::GetSourcePathByAssetId(assetId);
+            if (m_filterCallback(path))
+            {
+                AddPath(path);
+            }
+        }
```

Comment:
The filter is keeping unwanted files from being added. For all of my cases, I require the filter. There is no value in adding every file in the entire project for a single picking grid or drop down. Then again it could be a default behavior.